src/plugins/embeddable/public/lib/panel/panel_header/panel_header.tsx

```diff
@@ -165,6 +165,17 @@ export function PanelHeader({
         <EuiScreenReaderOnly>{getAriaLabel()}</EuiScreenReaderOnly>
       </div>
     );
+  } else if (hidePanelTitle) {
+    return (
+    <PanelOptionsMenu
+          getActionContextMenuPanel={getActionContextMenuPanel}
+          isViewMode={isViewMode}
+          closeContextMenu={closeContextMenu}
+          title={title}
+          index={index}
+        />
+        <EuiScreenReaderOnly>{getAriaLabel()}</EuiScreenReaderOnly>
+    );
```

Comment:
This is definitely getting there! :) Perhaps to avoid code duplication, you could assign `PanelOptionsMenu` to a variable instead - that will save quite a few lines. Or you could also conditionally render the wrapper - have you looked at  how Typescript's ternary operator works?

Just an FYI, I have a feeling that these changes will cause some of our functional tests to fail - so we might have to rethink how we gather our title visibilities in `panel_titles.ts`

Reply:
@Heenawter Hi, I made new changes to this PR. Would you please take a look at it?